proposed/logging-generator.md

```diff
@@ -0,0 +1,481 @@
+- `LogWithCustomEventName`: retrieve event name via `LoggerMessage` attribute. 
+- `LogWithDynamicLogLevel`: set log level dynamically, to allow log level to be set based on configuration input.
+- `LogWithNoTemplate`: auto-generate a message as a JSON blob if you don't supply one to output parameters.
+
+```csharp
+public partial class LoggingSample5
+{
+    private readonly ILogger _logger;
+
+    public LoggingSample5(ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    [LoggerMessage(1, LogLevel.Trace)]
+    public partial void LogEmptyMessage();
+
+    [LoggerMessage(9, LogLevel.Trace, "Fixed message", EventName = "CustomEventName")]
+    public partial void LogWithCustomEventName();
+
+    [LoggerMessage(10, "Welcome to {city} {province}!")]
+    public partial void LogWithDynamicLogLevel(string city, LogLevel level, string province);
+
+    [LoggerMessage(2, LogLevel.Trace)]
+    public partial void LogWithNoTemplate(string key1, string key2);
+
+    public void TestLogging()
+    {
+        LogEmptyMessage();
+        LogWithCustomEventName();
+        LogWithDynamicLogLevel("Vancouver", LogLevel.Warning, "BC");
+        LogWithDynamicLogLevel("Vancouver", LogLevel.Information, "BC");
+        LogWithNoTemplate("value2", "value2");
+    }
+}
+```
+output to `TestLogging()` using SimpleConsole:
+```
+trce: LoggingExample[1]
+      {}
+trce: LoggingExample[9]
+      Fixed message
+warn: LoggingExample[10]
+      Welcome to Vancouver BC!
+info: LoggingExample[10]
+      Welcome to Vancouver BC!
+trce: LoggingExample[2]
+      {"key1":"value2","key2":"value2"}
+```
+
+### Diagnostics
+
+This [gist](https://gist.github.com/maryamariyan/a1ab553bedb26b9886fbc2740ee9e954) shows 20 diagnostic messages that the generator can produce alongside use cases for each
+
+Using SYSLIBXXXX format as the diagnostic IDs. The gist also shows diagnostic categories against each sample.
+
+Q & A - The source generator
+
+- Question: Why would we want to enforce event ID uniqueness checks if existing log APIs still allow for using default event ID values? For example with `LogInformation` we have two overloads, one taking `EventId` and the other one not. But the proposal with `LoggerMessageAttribute` only requires taking an `EventId`.
+
+Answer: The enforced restriction added via `LoggerMessage` attribute aims at providing best practices for library authors more so than it does for app developers, who in most cases do not care about event IDs.
+
+This is a great feedback. In order to make this approach less restrictive, it would make sense to also allow `LoggerMessageAttribute` skip taking event IDs in combination with an analyzer that generates an error to recommend best practices while allowing app developers to cut corners for productivity. When turned off, the event IDs would default to zero or implicit event IDs would get generated.
+
+## Solution 2: using a custom string interpolation builder
```

Comment:
I wouldn't pose this as an either OR like I said before. Lets just list this as an optional feature we can do regardless of the source generator approach.

Reply:
## Supporting string interpolation builder in logging